packages/relay-runtime/network/RelayNetworkTypes.js

```diff
@@ -127,7 +116,8 @@ export type RequestResponseFunction = (
   variables: Variables,
   cacheConfig?: ?CacheConfig,
   uploadables?: UploadableMap,
-) => SyncOrPromise<RelayResponsePayload>;
+  observer: SingleObserver<RelayResponsePayload>,
+) => Disposable;
```

Comment:
This is the key change: it lets Relay runtime internally work with a uniform representation of a single, possibly async value, a SingleObserver. Only the RelayNetwork.create() helper has to know about Promises or other representations.